sql/atxs/atxs.go

```diff
@@ -863,7 +863,10 @@ func IterateAtxIdsWithMalfeasance(
 	return err
 }
 
-func PrevATXCollision(db sql.Executor, prev types.ATXID, id types.NodeID) (types.ATXID, types.ATXID, error) {
+// PrevATXCollisions returns all ATXs with the same prevATX as the given ATX ID from the same node ID.
+// It is used to detect double-publishing and double poet registrations.
+// The ATXs returned are ordered by received time so that the first one is the one that was seen first by the node.
+func PrevATXCollisions(db sql.Executor, prev types.ATXID, id types.NodeID) ([]types.ATXID, error) {
```

Comment:
The reason is the following: we want to get the earliest two ATXs (in terms of received, not in terms of public epoch) that have a collision, so that it doesn't matter when a collision is detected. This should lead to all nodes picking the same two ATXs for their proof, at least for nodes that are online near 100% of the time.

So theoretically it would be enough to only return the first two ATXs that are found. However we don't know if `handler_v1` calls this function or `handler_v2` (both can with the new malfeasance publisher). The first one must find two ATXs that are both v1, the second one prefers two ATXs that are v2 (even if an earlier v1 ATX also references the same prev ATX - that's just a fallback for when no v2 ATX is found).

I could build all that logic into this function but a) this own't work in a single query and b) should not be part of the SQL package.